RecoLocalTracker/SiPhase2VectorHitBuilder/plugins/SiPhase2RecHitMatcherESProducer.cc

```diff
@@ -0,0 +1,80 @@
+
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/ModuleFactory.h"
+#include "FWCore/Framework/interface/ESProducer.h"
+#include "RecoLocalTracker/Records/interface/TkPhase2OTCPERecord.h"
+
+#include "FWCore/Framework/interface/ESProducer.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "RecoLocalTracker/SiPhase2VectorHitBuilder/interface/VectorHitBuilderEDProducer.h"
+#include <memory>
+
+class SiPhase2RecHitMatcherESProducer : public edm::ESProducer {
+public:
+  SiPhase2RecHitMatcherESProducer(const edm::ParameterSet&);
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+  std::shared_ptr<VectorHitBuilderEDProducer> produce(const TkPhase2OTCPERecord&);
+
+private:
+  std::string name_;
+  std::shared_ptr<VectorHitBuilderEDProducer> matcher_;
+  edm::ParameterSet pset_;
+};
+
+SiPhase2RecHitMatcherESProducer::SiPhase2RecHitMatcherESProducer(const edm::ParameterSet& p) {
+  name_ = p.getParameter<std::string>("ComponentName");
+  pset_ = p;
+  setWhatProduced(this, name_);
+}
+
+std::shared_ptr<VectorHitBuilderEDProducer> SiPhase2RecHitMatcherESProducer::produce(
+    const TkPhase2OTCPERecord& iRecord) {
+  if (name_ == "SiPhase2VectorHitMatcher") {
```

Comment:
Why? What would be the point of having this ESProducer in the configuration with a name that leads to ~nothing happening?

Reply:
Good question. @ebrondol , was there any thought of having a different option here at some point? If not I think we can remove the if clause.

Reply:
Yes, there was but it can be done in many ways and this is indeed a more troublesome one!